atomic_reactor/plugins/pre_pin_operator_digest.py

```diff
@@ -224,13 +249,17 @@ def _get_replacement_pullspecs(self, pullspecs):
 
             self.log.info("Final pullspec: %s", replaced)
 
-            if replaced != original:
-                replacements[original] = replaced
+            replacements.append({
+                'original': original,
+                'new': replaced,
+                'pinned': pinned,
+                'replaced': replaced != original
+            })
 
         replacement_lines = "\n".join(
-            "{} -> {}".format(p, replacements[p]) if p in replacements
-            else "{} - no change".format(p)
-            for p in pullspecs
+            "{} -> {}".format(r['original'], r['new']) if r['replaced']
```

Comment:
Consider `"{original} -> {new}".format(**r)` and `"{original} - no change".format(**r)`

Reply:
updated